org.abapgit.adt.backend/src/org/abapgit/adt/backend/model/abapgitrepositories/impl/RepositoryImpl.java

```diff
@@ -0,0 +1,1066 @@
+
+import org.eclipse.emf.common.notify.Notification;
+import org.eclipse.emf.common.notify.NotificationChain;
+
+import org.eclipse.emf.common.util.EList;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.InternalEObject;
+
+import org.eclipse.emf.ecore.impl.ENotificationImpl;
+import org.eclipse.emf.ecore.impl.MinimalEObjectImpl;
+
+import org.eclipse.emf.ecore.util.EObjectContainmentEList;
+import org.eclipse.emf.ecore.util.InternalEList;
+
+/**
+ * <!-- begin-user-doc -->
+ * An implementation of the model object '<em><b>Repository</b></em>'.
+ * <!-- end-user-doc -->
+ * <p>
+ * The following features are implemented:
+ * </p>
+ * <ul>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getLinks <em>Links</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getKey <em>Key</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getPackage <em>Package</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getUrl <em>Url</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getBranchName <em>Branch Name</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getCreatedBy <em>Created By</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getCreatedEmail <em>Created Email</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getCreatedAt <em>Created At</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getDeserializedAt <em>Deserialized At</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getDeserializedEmail <em>Deserialized Email</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getDeserializedBy <em>Deserialized By</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getStatus <em>Status</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getStatusText <em>Status Text</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getRemoteUser <em>Remote User</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getRemotePassword <em>Remote Password</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getFirstCommit <em>First Commit</em>}</li>
+ *   <li>{@link org.abapgit.adt.backend.model.abapgitrepositories.impl.RepositoryImpl#getTransportRequest <em>Transport Request</em>}</li>
+ * </ul>
+ *
+ * @generated
+ */
+public class RepositoryImpl extends MinimalEObjectImpl.Container implements IRepository {
```

Comment:
Provide an implementation for the **equals and hashcode** method for the Repository object.
Also, do the same for other objects like AbapObjectImpl...

Reply:
Provided the hashcode and equals method.